esp-hal/src/dma/buffers.rs

```diff
@@ -574,24 +800,48 @@ impl DmaRxTxBuf {
         tx_descriptors: &'static mut [DmaDescriptor],
         buffer: &'static mut [u8],
     ) -> Result<Self, DmaBufError> {
-        if !is_slice_in_dram(buffer) {
-            return Err(DmaBufError::UnsupportedMemoryRegion);
-        }
-
         let mut buf = Self {
             rx_descriptors: DescriptorSet::new(rx_descriptors)?,
             tx_descriptors: DescriptorSet::new(tx_descriptors)?,
             buffer,
+            burst: BurstConfig::default(),
         };
-        buf.rx_descriptors
-            .link_with_buffer(buf.buffer, max_chunk_size(None))?;
-        buf.tx_descriptors
-            .link_with_buffer(buf.buffer, max_chunk_size(None))?;
-        buf.set_length(buf.capacity());
+
+        let capacity = buf.capacity();
+        buf.configure(buf.burst, capacity)?;
 
         Ok(buf)
     }
 
+    fn configure(&mut self, burst: BurstConfig, length: usize) -> Result<(), DmaBufError> {
+        burst.ensure_buffer_compatible(self.buffer, TransferDirection::In)?;
+        burst.ensure_buffer_compatible(self.buffer, TransferDirection::Out)?;
+
+        self.rx_descriptors.link_with_buffer(
+            self.buffer,
+            burst.max_chunk_size_for(self.buffer, TransferDirection::In),
+        )?;
+        self.tx_descriptors.link_with_buffer(
+            self.buffer,
+            burst.max_chunk_size_for(self.buffer, TransferDirection::Out),
+        )?;
+
+        self.set_length_fallible(length, burst)?;
+        self.burst = burst;
+
+        Ok(())
+    }
+
+    /// Configures the DMA to use burst transfers to access this buffer.
+    ///
+    /// Note that the hardware is allowed to ignore this setting. If you attempt
+    /// to use burst transfers with improperly aligned buffers, starting the
+    /// transfer will result in [`DmaError::InvalidAlignment`].
```

Comment:
It will?

Reply:
No longer because someone was unhappy with perf.